source/unix/openssl_platform_init.c

```diff
@@ -302,68 +136,20 @@ static int s_resolve_libcrypto_symbols(enum aws_libcrypto_version version, void
     return found_version;
 }
 
-static int s_resolve_libcrypto_version(enum aws_libcrypto_version version) {
-    const char *libcrypto_102 = "libcrypto.so.1.0.0";
-    const char *libcrypto_111 = "libcrypto.so.1.1";
-    switch (version) {
-        case AWS_LIBCRYPTO_NONE: {
-            FLOGF("searching process and loaded modules");
-            void *process = dlopen(NULL, RTLD_NOW);
-            AWS_FATAL_ASSERT(process && "Unable to load symbols from process space");
-            int result = s_resolve_libcrypto_symbols(version, process);
-            dlclose(process);
-            return result;
-        }
-        case AWS_LIBCRYPTO_1_0_2: {
-            FLOGF("loading libcrypto 1.0.2");
-            void *module = dlopen(libcrypto_102, RTLD_NOW);
-            if (module) {
-                FLOGF("resolving against libcrypto 1.0.2");
-                int result = s_resolve_libcrypto_symbols(version, module);
-                dlclose(module);
-                return result;
-            }
-            FLOGF("libcrypto 1.0.2 not found");
-            break;
-        }
-        case AWS_LIBCRYPTO_1_1_1: {
-            FLOGF("loading libcrypto 1.1.1");
-            void *module = dlopen(libcrypto_111, RTLD_NOW);
-            if (module) {
-                FLOGF("resolving against libcrypto 1.1.1");
-                int result = s_resolve_libcrypto_symbols(version, module);
-                dlclose(module);
-                return result;
-            }
-            FLOGF("libcrypto 1.1.1 not found");
-            break;
-        }
-        default:
-            AWS_FATAL_ASSERT("Attempted to use an unsupported version of libcrypto");
-    }
-    return AWS_LIBCRYPTO_NONE;
-}
-
 static int s_resolve_libcrypto(void) {
     if (s_libcrypto_version != AWS_LIBCRYPTO_NONE) {
         return s_libcrypto_version;
     }
 
     /* Try to auto-resolve against what's linked in/process space */
     FLOGF("Attempting auto-resolve against static linkage");
-    s_libcrypto_version = s_resolve_libcrypto_version(AWS_LIBCRYPTO_NONE);
-    /* try 1.0.2 */
-    if (s_libcrypto_version == AWS_LIBCRYPTO_NONE) {
-        FLOGF("Attempting resolve against libcrypto 1.0.2");
-        s_libcrypto_version = s_resolve_libcrypto_version(AWS_LIBCRYPTO_1_0_2);
-    }
-    /* try 1.1.1 */
-    if (s_libcrypto_version == AWS_LIBCRYPTO_NONE) {
-        FLOGF("Attempting resolve against libcrypto 1.1.1");
-        s_libcrypto_version = s_resolve_libcrypto_version(AWS_LIBCRYPTO_1_1_1);
-    }
+    FLOGF("searching process and loaded modules");
```

Comment:
The line above is not necessary anymore since we aren't doing multi-pass resolution